test/test_fastsim.cpp

```diff
@@ -73,3 +73,14 @@ BOOST_AUTO_TEST_CASE(CPUSort)
     BOOST_CHECK_EQUAL(scores[3], 3);
 
 }
+
+BOOST_AUTO_TEST_CASE(FoldFingerprint)
+{
+    int factor = 2;
+    vector<int> fp = {32, 24, 11, 7};
+    vector<int> ref_answer = {43, 31};
+    vector<int> answer = fold_fingerprint(fp, factor);
```

Comment:
@Wisakedjack - can you add another example in this test case for folding by factor of 4?

Reply:
Added another test to fold by factor 4.